src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/cache/FieldDataCacheRca.java

```diff
@@ -0,0 +1,204 @@
+            ResourceContext context;
+            HotNodeSummary nodeSummary;
+
+            ClusterDetailsEventProcessor.NodeDetails currentNode = ClusterDetailsEventProcessor.getCurrentNodeDetails();
+            double cacheSize = getTotalSizeInMB(fieldDataCacheSize);
+            double cacheMaxSize = getTotalSizeInMB(fieldDataCacheMaxSize);
+            exceedsSize = cacheMaxSize != 0 && cacheMaxSize != 0 && cacheSize > cacheMaxSize;
+            if (cacheEvictionCollector.isUnhealthy(currTimestamp) && exceedsSize) {
+                context = new ResourceContext(Resources.State.UNHEALTHY);
+                nodeSummary = new HotNodeSummary(currentNode.getId(), currentNode.getHostAddress());
+                nodeSummary.appendNestedSummary(cacheEvictionCollector.generateSummary(currTimestamp));
+            }
+            else {
+                context = new ResourceContext(Resources.State.HEALTHY);
+                nodeSummary = null;
+            }
+
+            counter = 0;
+            exceedsSize = Boolean.FALSE;
+            return new ResourceFlowUnit<>(currTimestamp, context, nodeSummary, !currentNode.getIsMasterNode());
+        }
+        else {
+            return new ResourceFlowUnit<>(currTimestamp);
+        }
+    }
+
+    @Override
+    public void generateFlowUnitListFromWire(FlowUnitOperationArgWrapper args) {
+        final List<FlowUnitMessage> flowUnitMessages =
+                args.getWireHopper().readFromWire(args.getNode());
+        List<ResourceFlowUnit<HotNodeSummary>> flowUnitList = new ArrayList<>();
+        LOG.debug("rca: Executing fromWire: {}", this.getClass().getSimpleName());
+        for (FlowUnitMessage flowUnitMessage : flowUnitMessages) {
+            flowUnitList.add(ResourceFlowUnit.buildFlowUnitFromWrapper(flowUnitMessage));
+        }
+        setFlowUnits(flowUnitList);
+    }
+
+    /**
+     * A collector class to collect eviction metrics
+     */
+    private static class CacheEvictionCollector {
+        private final Resource cache;
+        private final Metric cacheEvictionMetrics;
+        private boolean hasEvictions;
+        private long evictionTimestamp;
+        private long evictionTimePeriodThreshold;
+
+        private CacheEvictionCollector(final Resource cache, final Metric cacheEvictionMetrics,
+                                       final long threshold) {
+            this.cache = cache;
+            this.cacheEvictionMetrics = cacheEvictionMetrics;
+            this.hasEvictions = false;
+            this.evictionTimestamp = 0;
+            this.evictionTimePeriodThreshold = threshold;
+        }
+
+        public void collect(final long currTimestamp) {
+            for (MetricFlowUnit flowUnit : cacheEvictionMetrics.getFlowUnits()) {
+                if (flowUnit.isEmpty() || flowUnit.getData() == null) {
+                    continue;
+                }
+
+                double evictionCount = flowUnit.getData().stream().mapToDouble(
```

Comment:
`flowUnit.getData().stream()` -- Can each flowUnit here have multiple data points?

Reply:
Got it, so we need to sum the stream because this is a shard level metric?